libcudacxx/include/cuda/__annotated_ptr/access_property_encoding.h

```diff
@@ -230,116 +234,116 @@ struct __block_desc_t
   uint64_t __l2_sector_promote_256B           : 1;
   uint64_t __ap_reserved3                     : 1;
 
-  _CCCL_HOST_DEVICE constexpr std::uint64_t __get_descriptor_cexpr() const noexcept
+  [[nodiscard]] _CCCL_HOST_DEVICE constexpr uint64_t __get_descriptor_cexpr() const noexcept
   {
-    return std::uint64_t(__ap_reserved) << 0 | std::uint64_t(__block_count) << 37 | std::uint64_t(__block_start) << 44
-         | std::uint64_t(__ap_reserved2) << 51 | std::uint64_t(__block_size) << 52 | std::uint64_t(__l2_cop_off) << 56
-         | std::uint64_t(__l2_cop_on) << 57 | std::uint64_t(__l2_descriptor_mode) << 59
-         | std::uint64_t(__l1_inv_dont_allocate) << 61 | std::uint64_t(__l2_sector_promote_256B) << 62
-         | std::uint64_t(__ap_reserved3) << 63;
+    return static_cast<uint64_t>(__ap_reserved) << 0 | //
+           static_cast<uint64_t>(__block_count) << 37 | //
+           static_cast<uint64_t>(__block_start) << 44 | //
+           static_cast<uint64_t>(__ap_reserved2) << 51 | //
+           static_cast<uint64_t>(__block_size) << 52 | //
+           static_cast<uint64_t>(__l2_cop_off) << 56 | //
+           static_cast<uint64_t>(__l2_cop_on) << 57 | //
+           static_cast<uint64_t>(__l2_descriptor_mode) << 59 | //
+           static_cast<uint64_t>(__l1_inv_dont_allocate) << 61 | //
+           static_cast<uint64_t>(__l2_sector_promote_256B) << 62 | //
+           static_cast<uint64_t>(__ap_reserved3) << 63; //
```

Comment:
Can we move the `|` to the left? Its much easier to read if the operator is on the left side